### PR TITLE
Add framework support policy

### PR DESCRIPTION
**Pull request by nandab-work**
*Issue #, if available:*

*Description of changes:*
- Add Framework Support Policy
- Supported images file is being deleted as this information is being added to the support policy file.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by aws-tianquaw — COMMENTED** [review]
Do we maintain this file manually or is there a way to automatically update it? For example, the unsupported versions need to be removed from the table 1 year after their EOD date. How should we ensure that we will make such updates in time?

File: support_policy.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+| 0.10.x        | public.ecr.aws/sagemaker/sagemaker-distribution:0.10-gpu | June 6th, 2024   |
+| 0.9.x         | public.ecr.aws/sagemaker/sagemaker-distribution:0.9-gpu  | May 3rd, 2024    |
+| 0.8.x         | public.ecr.aws/sagemaker/sagemaker-distribution:0.8-gpu  | April 9th, 2024  |
+| 0.7.x         | public.ecr.aws/sagemaker/sagemaker-distribution:0.7-gpu  | April 5th, 2024  |
+
+## Frequently Asked Questions
+
+**[Q]** Can I still use older images after an image is no longer supported?
+
+**[A]** Yes, older images remain available in ECR after they reach end of support. However, we highly recommend upgrading to a supported image version that are continuously receiving security updates and bug fixes. It is the customer's responsibility to manage any vulnerabilities that arise due to choosing an image version that is no longer supported by AWS.
```
Comment (aws-tianquaw) [inline support_policy.md:75]: Should we link the shared responsibility model doc? https://aws.amazon.com/compliance/shared-responsibility-model/

File: support_policy.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+| Version | Description | Release Cadence |
+| :---:   | :---:       | :---:           |
+| Major   | Amazon SageMaker Distribution's major version releases involve upgrading all of its core dependencies to the latest compatible versions. These major releases may also add or remove packages as part of the update. Major versions are denoted by the first number in the version string, such as 1.0, 2.0, or 3.0. | 6 months |
+| Minor   | Amazon SageMaker Distribution's minor version releases include upgrading all of its core dependencies to the latest compatible minor versions within the same major version. SageMaker Distribution can add new packages during a minor version release. Minor versions are denoted by the second number in the version string, for example, 1.1, 1.2, or 2.1. | 1 month |
```
Comment (aws-tianquaw) [inline support_policy.md:10]: As I understand, we are not currently following 1-month, 6-month schedule to release minor/major versions. Once they are added to the policy, does it mean we have to follow this schedule? @balajisankar15 Could you verify if such schedules are valid and should be made public?

**Comment by nandab-work** [comment]
> Do we maintain this file manually or is there a way to automatically update it? For example, the unsupported versions need to be removed from the table 1 year after their EOD date. How should we ensure that we will make such updates in time?

This file would be maintained manually for now, and updating this file would be a part of our Image release and deprecation SOP. Entries would be removed from this table during one of these two steps. Though the vision is to automate this as part of our image release and deprecation process.

File: support_policy.md (hunk 1 above)
Comment (nandab-work) [reply support_policy.md:75]: Yes, I can link this doc.

**Review by balajisankar15 — DISMISSED** [review]

**Review by aws-tianquaw — DISMISSED** [review]

**Review by ketan-vijayvargiya — APPROVED** [review]

**Review by aws-tianquaw — APPROVED** [review]

**Review by balajisankar15 — APPROVED** [review]

**Pull request merged**